ui/tests/acceptance/allocation-detail-test.js

```diff
@@ -76,37 +76,58 @@ module('Acceptance | allocation detail', function(hooks) {
-    const task = server.db.taskStates.where({ allocationId: allocation.id }).sortBy('name')[0];
-    const taskResources = allocation.taskResourceIds
-      .map(id => server.db.taskResources.find(id))
-      .sortBy('name')[0];
-    const reservedPorts = taskResources.resources.Networks[0].ReservedPorts;
-    const dynamicPorts = taskResources.resources.Networks[0].DynamicPorts;
-    const taskRow = Allocation.tasks.objectAt(0);
-    const events = server.db.taskEvents.where({ taskStateId: task.id });
-    const event = events[events.length - 1];
-
-    assert.equal(taskRow.name, task.name, 'Name');
-    assert.equal(taskRow.state, task.state, 'State');
-    assert.equal(taskRow.message, event.displayMessage, 'Event Message');
-    assert.equal(
-      taskRow.time,
-      moment(event.time / 1000000).format("MMM DD, 'YY HH:mm:ss ZZ"),
-      'Event Time'
-    );
-
-    assert.ok(reservedPorts.length, 'The task has reserved ports');
-    assert.ok(dynamicPorts.length, 'The task has dynamic ports');
-
-    const addressesText = taskRow.ports;
-    reservedPorts.forEach(port => {
-      assert.ok(addressesText.includes(port.Label), `Found label ${port.Label}`);
-      assert.ok(addressesText.includes(port.Value), `Found value ${port.Value}`);
-    });
-    dynamicPorts.forEach(port => {
-      assert.ok(addressesText.includes(port.Label), `Found label ${port.Label}`);
-      assert.ok(addressesText.includes(port.Value), `Found value ${port.Value}`);
-    });
+    server.db.taskStates
+      .where({ allocationId: allocation.id })
+      .sortBy('name')
+      .forEach((task, taskIndex) => {
+        const taskResources = allocation.taskResourceIds
+          .map(id => server.db.taskResources.find(id))
+          .sortBy('name')[taskIndex];
+        const network = taskResources.resources.Networks[0];
+        const reservedPorts = network.ReservedPorts;
+        const dynamicPorts = network.DynamicPorts;
+        const taskRow = Allocation.tasks.objectAt(taskIndex);
+        const events = server.db.taskEvents.where({ taskStateId: task.id });
+        const event = events[events.length - 1];
+
+        assert.equal(taskRow.name, task.name, 'Name');
+        assert.equal(taskRow.state, task.state, 'State');
+        assert.equal(taskRow.message, event.displayMessage, 'Event Message');
+        assert.equal(
+          taskRow.time,
+          moment(event.time / 1000000).format("MMM DD, 'YY HH:mm:ss ZZ"),
+          'Event Time'
+        );
+
+        assert.ok(reservedPorts.length, 'The task has reserved ports');
+        assert.ok(dynamicPorts.length, 'The task has dynamic ports');
+
+        reservedPorts.forEach((serverPort, index) => {
+          taskRow.reservedPorts[index].as(renderedPort => {
+            assert.equal(renderedPort.label, `${serverPort.Label}:`);
+            assert.equal(renderedPort.link.text, `${network.IP}:${serverPort.Value}`);
+
+            if (network.IP.includes(':')) {
```

Comment:
Can we use the same `is-ip` here as well?

Reply:
Definitely, I added the test before implementing the fix, so I forgot to update this, but done now, thanks!